code/modules/multiz/structures.dm

```diff
@@ -0,0 +1,281 @@
+//////////////////////////////
+//Contents: Ladders, Stairs.//
+//////////////////////////////
+
+/obj/structure/ladder
+	name = "ladder"
+	desc = "A ladder. You can climb it up and down."
+	icon_state = "ladder01"
+	icon = 'icons/obj/structures.dmi'
+	density = 0
+	opacity = 0
+	anchored = 1
+
+	var/allowed_directions = DOWN
+	var/obj/structure/ladder/target_up
+	var/obj/structure/ladder/target_down
+
+	var/const/climb_time = 2 SECONDS
+
+/obj/structure/ladder/initialize()
+	// the upper will connect to the lower
+	if(allowed_directions & DOWN) //we only want to do the top one, as it will initialize the ones before it.
+		for(var/obj/structure/ladder/L in GetBelow(src))
+			if(L.allowed_directions & UP)
+				target_down = L
+				L.target_up = src
+				return
+	update_icon()
+
+/obj/structure/ladder/Destroy()
+	if(target_down)
+		target_down.target_up = null
+		target_down = null
+	if(target_up)
+		target_up.target_down = null
+		target_up = null
+	return ..()
+
+/obj/structure/ladder/attackby(obj/item/C as obj, mob/user as mob)
+	attack_hand(user)
+	return
```

Comment:
trailing return

We do need to be semi careful about how we allow ladders to be built (if we do, like stairs) to prevent laddering into the armory or something (special floors?)

ladders on a lower Z-level should prevent falling, allowing you to ladder up into an open space above

Reply:
Fixed trailing return